### PR TITLE
Support for TipUp help boxes

### PR DESCRIPTION
**Pull request by ppacher**

### CODE REVIEW

**Comment by davegson** [comment]
the texts are ready to go from my side - it would be nice if either @ppacher or @dhaavi could have a look so the new texts are reviewed too.

I reduced the amount of tip ups since (A) it is not part of the specification and (B) it would have taken up another chunk of my work week with more pressing things to do imo. Happy to discuss this with @dhaavi when you get to it

I only moved the unused texts to a temporary .yaml file, the tip ups in the code remain to be removed. Probably best if you do that @ppacher

File: modules/portmaster/src/i18n/helptexts.yaml [hunk 1]
```diff
@@ -41,95 +40,92 @@ uiMode:
 pilot-widget-NetworkRating:
   title: Network Rating
   content: |
-    Some features and settings of the Portmaster behave differently, depending on the environment.
+    Control your privacy even when connecting to new networks.
```
Comment (dhaavi) [inline modules/portmaster/src/i18n/helptexts.yaml:44]: The network does not have to be new - situations can change while being connected.

File: modules/portmaster/src/i18n/helptexts.yaml [hunk 2]
```diff
@@ -41,95 +40,92 @@ uiMode:
...
 
-    In order to help the Portmaster provide the most adequate protection, please select "Trusted", "Untrusted" or "Danger" according to the situation you are in.<br>
-    Continue to learn about these different levels.
+    In the Portmaster you configure settings to be active in one environment but not in the other, like allowing sensitive connections at home but not at the public library.
+
+    The only thing you have to do is to change the network rating whenever you connect to a different network.
```
Comment (dhaavi) [inline modules/portmaster/src/i18n/helptexts.yaml:48]: As mentioned earlier, this may change while being connected.

File: modules/portmaster/src/i18n/helptexts.yaml [hunk 3]
```diff
@@ -41,95 +40,92 @@ uiMode:
...
+     - public WiFi of a coffeeshop, a library, a train, a hotel, ...
+     - network of a non-tech-savvy relative
   nextKey: pilot-widget-NetworkRating-Danger
 
 pilot-widget-NetworkRating-Danger:
-  title: Danger Rating
+  title: "Network Rating: Danger"
   content: |
-    This rating is for emergencies, when you have a reasonable suspicion that the network is hacked or otherwise hostile to you.
+    You think that the current network is hacked or otherwise hostile to you.
```
Comment (dhaavi) [inline modules/portmaster/src/i18n/helptexts.yaml:74]: I think the correct wording is "hostile towards you".

File: modules/portmaster/src/i18n/helptexts.yaml [hunk 4]
```diff
@@ -41,95 +40,92 @@ uiMode:
...
 pilot-widget-NetworkRating-Trusted:
-  title: Trusted Rating
+  title: "Network Rating: Trusted"
   content: |
-    This rating is for when you expect a network to be secure and help protect you.<br>
-    In most cases, you will use this for your network at home or at friends.
+    You trust the current network to be secure and protect you.
+
+    Examples:
+     - your home network
```
Comment (dhaavi) [inline modules/portmaster/src/i18n/helptexts.yaml:57]: I would recommend normal sentence casing and punctuation.

File: modules/portmaster/src/i18n/helptexts.yaml [hunk 5]
```diff
@@ -41,95 +40,92 @@ uiMode:
...
-    The setting is off for all network rating levels.
-    (Note: Not all network level settings can be fully turned off in
-    order to provide a protective baseline for the danger level)
+    Choose in which environment(s) this setting should be enforced.
+
+    Easily configure settings for different environments. When changing the "Network Rating" in the top left all settings will immediately adjust to your preferences.
+
+    - **Off**: This setting is completely turned off and not enforced anywhere
+    - **Danger**: This setting is only enforced when you set your Network Rating to "Danger"
+    - **Untrusted**: This setting is both enforced in the "Danger" _AND_ in the "Untrusted" environment
```
Comment (dhaavi) [inline modules/portmaster/src/i18n/helptexts.yaml:124]: I feel "enforced" is too hard. It's just enabled. A global setting will not be enforced on a per-app level.

File: modules/portmaster/src/i18n/helptexts.yaml [hunk 6]
```diff
@@ -1,22 +1,23 @@
...
+#     It even supports markdown features like:
+#     - order lists
+#     - with multiple items
+# 
+#     And :rocket: emojis
+# 
+#     ### :tada: :facepalm:
+#   url: https://docs.safing.io
+#   urlText: Show me!
+#   nextKey: navMonitor
```
Comment (dhaavi) [inline modules/portmaster/src/i18n/helptexts.yaml]: The example is only displayed in the developer mode (Due to being added to the add widget box) and is great for testing things. Would rather leave it in there.

Comment (dhaavi) [inline modules/portmaster/src/i18n/helptexts.yaml]: The example is only displayed in the developer mode (Due to being added to the add widget box) and is great for testing things. Would rather leave it in there.

File: modules/portmaster/src/i18n/helptexts.yaml [hunk 7]
```diff
@@ -119,7 +119,7 @@ networkRatingSwitch:
 
     Easily configure settings for different environments. When changing the "Network Rating" in the top left all settings will immediately adjust to your preferences.
 
-    - **Off**: This setting is completely turned off and not enforced anywhere
+    - **Off**: This setting is completely turned off and is never enforced
```
Comment (dhaavi) [inline modules/portmaster/src/i18n/helptexts.yaml:122]: See earlier comment about enforcing.

File: modules/portmaster/src/app/widgets/pilot-widget/pilot-widget.html [hunk 8]
```diff
@@ -21,8 +21,8 @@ <h2>{{lockLevel?.displayText}}</h2>
 
 <div tipUpAnchor class="level-list" [class.auto-pilot]="mode === 'auto'">
   <div class="rate-header">
-    <label style="white-space: nowrap; margin-right: 5px">
-      Rate your network
+    <label style="white-space: nowrap; margin-right: 5px; opacity: 0.9;">
+      Network Rating:
       <app-tipup key="pilot-widget-NetworkRating"></app-tipup>
     </label>
     <app-dropdown *appExpertiseLevel="'developer'" [ngModel]="mode" (ngModelChange)="updateMode($event)">
```
Comment (dhaavi) [inline modules/portmaster/src/app/widgets/pilot-widget/pilot-widget.html]: Change is out of scope for review. Needs discussion.

File: modules/portmaster/src/i18n/helptexts.yaml [hunk 9]
```diff
@@ -21,61 +21,6 @@ myKey:
 ##############
 ### Navigation
 
-intro:
```
Comment (dhaavi) [inline modules/portmaster/src/i18n/helptexts.yaml]: Scope change is out of scope for review. Needs discussion.

**Review by dhaavi — CHANGES_REQUESTED** [review]
Some minor wording feedback. Out of scope changes need discussion.

File: modules/portmaster/src/i18n/helptexts.yaml (hunk 1 above)
Comment (davegson) [reply modules/portmaster/src/i18n/helptexts.yaml:110]: I know that this is not technically correct since there are other potential scenarios. However, we are explaining this feature step by step. And for the first step teaching users to _always_ check this when connecting to a new network is a good thing. Later down the road (@Danger) they learn that this can even be done when being in the same network.

File: modules/portmaster/src/i18n/helptexts.yaml (hunk 2 above)
Comment (davegson) [reply modules/portmaster/src/i18n/helptexts.yaml:114]: same point applies as mentioned in the other discussion

File: modules/portmaster/src/i18n/helptexts.yaml (hunk 4 above)
Comment (davegson) [reply modules/portmaster/src/i18n/helptexts.yaml:123]: what do you mean hear? What would such a recommended sentence be?

Comment (davegson) [reply modules/portmaster/src/i18n/helptexts.yaml:123]: I saw a use case provided by you (the shield colors) and disagree from a typography perspective. This seems very uncommon and we rarely use it too, think docs homepage or [linux page](https://docs.safing.io/portmaster/install/linux). Remove the dots there to unify this.

File: modules/portmaster/src/i18n/helptexts.yaml (hunk 1 above)
Comment (dhaavi) [reply modules/portmaster/src/i18n/helptexts.yaml:110]: I see. Ok. Let's watch out for feedback regarding this.

File: modules/portmaster/src/i18n/helptexts.yaml (hunk 4 above)
Comment (dhaavi) [reply modules/portmaster/src/i18n/helptexts.yaml:123]: Ok. I just think it's weird.

**Review by dhaavi — CHANGES_REQUESTED** [review]
Wording is great! 👌
Only waiting for some final minor fixes from @ppacher.

**Review by dhaavi — DISMISSED** [review]
Yay!

**Review by dhaavi — APPROVED** [review]
Conflict resolving.
